rules/DeadCode/Rector/ClassMethod/RemoveUnusedPrivateMethodRector.php

```diff
@@ -228,4 +235,31 @@ private function shouldSkipArrayCallable(Class_ $class, ?ArrayCallable $arrayCal
         // is current class method?
         return ! $this->isName($class, $arrayCallable->getClass());
     }
+
+    private function isPrivateAbstractMethodInTrait(ClassMethod $classMethod, ?string $classMethodName): bool
+    {
+        if (null === $classMethodName) {
+            return false;
+        }
+
+        $scope = $classMethod->getAttribute(AttributeKey::SCOPE);
+        if (! $scope instanceof Scope) {
+            return false;
+        }
+
+        $classReflection = $scope->getClassReflection();
+        if (null === $classReflection) {
+            return false;
+        }
+
+        $traits = $this->astResolver->parseClassReflectionTraits($classReflection);
+        foreach ($traits as $trait) {
+            $method = $trait->getMethod($classMethodName);
+            if ($method instanceof ClassMethod) {
+                return true;
```

Comment:
I am not in front of PC, should it return $method->isAbstract() ?

Reply:
True, if the method would be private but not abstract that wouldn't cover the problem. I'll add a test case and fix this